docs/latest/examples/authentication-with-supabase.md

```diff
@@ -0,0 +1,217 @@
+      const { error } = await supabase.auth.verifyOtp({ type, token_hash });
+      if (error) throw error;
+    }
+
+    redirectTo.searchParams.delete("next");
+    return Response.redirect(redirectTo);
+  },
+};
+```
+
+Have a look at the Supabase docs on the [details on how to configure e-mail templates and other endpoints](https://supabase.com/docs/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr) like `/password-reset` you would need for a full implementation.
+
+
+## Sign In
+
+The `/api/sign-in` route is pretty straight-forward, too:
+
+```ts routes/api/sign-in.ts
+export const handler: Handlers = {
+  async POST(req) {
+    const form = await req.formData();
+    const email = String(form.get("email"));
+    const password = String(form.get("password"));
+
+    const headers = new Headers();
+    headers.set("location", "/");
+
+    const supabase = createSupabaseClient(req, headers);
+    const { error } = await supabase.auth.signInWithPassword({ email, password });
+
+    if (error) throw error;
+
+    return new Response(null, { status t 303, headers });
+  },
+};
+```
+
+Note that we're passing `headers` this time. The Supabase client will set the session as cookie for us which we will want to pick up in the middleware that we are writing next.
+
+## Middleware
+
+We can now write a middleware that will check the auth status of any request, guarding any protected routes. You can read up on middlewares and where to put them [in the docs](https://fresh.deno.dev/docs/concepts/middleware).
+
+```ts routes/_middleware.ts
+export const handler = [
+  async function authMiddleware(req: Request, ctx: FreshContext) {
+    const url = new URL(req.url);
+    const headers = new Headers();
+    headers.set("location", "/");
+
+    const supabase = createSupabaseClient(req, headers);
+    // NOTE: Always use `getUser` instead of `getSession` as this calls the Supabase API and revalidates the token!
+    const { error, data: { user } } = await supabase.auth.getUser();
+
+    const isProtectedRoute = url.pathname.includes("secret");
+
+    // Don't mind 401 as it just means no credentials were provided, e.g. there was no session cookie.
+    if (error && error.status !== 401) throw error;
+
+    if (isProtectedRoute && !user) {
+      return new Response(null, { status: 303, headers });
+    }
+
+    ctx.state.user = user;
+
+    return ctx.next();
+  },
+];
+```
+
+That's it! These are the building blocks for implementing the PKCE authentication flow in a Fresh app using Supabase.
+
+## Closing Notes
+
+The purpose of the example app we built here is to showcase the basic building blocks of an implementation. As such, it is limited in functionality and purposefully leaves out things like [password reset](https://supabase.com/docs/guides/auth/server-side/email-based-auth-with-pkce-flow-for-ssr), [error handling](https://fresh.deno.dev/docs/concepts/error-pages) and passing server state to components.
```

Comment:
This is good info, but perhaps it should be moved to the start of the tutorial so the reader is aware of such missing details. I'd mention that form validation (i.e. checking that inputs are populated) logic is omitted too.